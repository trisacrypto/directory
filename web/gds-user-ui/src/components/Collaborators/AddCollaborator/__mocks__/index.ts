
export const collaboratorMockValue: any = {
    data: [
        {
            id: "1",
            name: "test",
            email: "test@local.dev",
            roles: ["admin"],
            created_at: "2021-01-01",
        },
    ],
};

export const userOrganizationMockValue: any = {
    data: {
        id: "1",
        name: "test",
        domain: "test.local.dev",
        created_at: "2021-01-01",
    },
};
