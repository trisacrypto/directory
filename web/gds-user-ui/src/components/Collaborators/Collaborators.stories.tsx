import { Meta, Story } from "@storybook/react";
import Collaborators from ".";

export default {
  title: "components/Collaborators",
  component: Collaborators,
} as Meta;

const Template: Story = (args) => <Collaborators {...args} />;

export const Standard = Template.bind({});
Standard.args = {};
