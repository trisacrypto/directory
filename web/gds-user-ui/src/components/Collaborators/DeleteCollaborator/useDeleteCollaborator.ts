import { useMutation } from '@tanstack/react-query';
import { deleteCollaborator as deleteCollaboratorService } from 'modules/dashboard/collaborator/CollaboratorService';
import type { DeleteCollaboratorMutation } from 'components/Collaborators/DeleteCollaborator/DeleteCollaboratorType';

export function useDeleteCollaborator(): DeleteCollaboratorMutation {
    const mutation = useMutation(['deleteCollaborator'], deleteCollaboratorService);
    return {
        deleteCollaborator: mutation.mutate,
        reset: mutation.reset,
        collaborator: mutation.data,
        hasCollaboratorFailed: mutation.isError,
        wasCollaboratorDeleted: mutation.isSuccess,
        isDeleting: mutation.isLoading,
        errorMessage: mutation.error,
    };
}

