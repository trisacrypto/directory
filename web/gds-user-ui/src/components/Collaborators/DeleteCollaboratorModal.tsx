import {
  Box,
  Button,
  Checkbox,
  Link,
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
  Text,
  useDisclosure,
  VStack
} from '@chakra-ui/react';
import { Trans } from '@lingui/macro';
import { BsTrash } from 'react-icons/bs';

function DeleteCollaboratorModal() {
  const { isOpen, onOpen, onClose } = useDisclosure();

  return (
    <Link color="blue" onClick={onOpen}>
      <BsTrash fontSize="26px" />
      <Modal isOpen={isOpen} onClose={onClose}>
        <ModalOverlay />
        <ModalContent>
          <ModalHeader textTransform="capitalize" textAlign="center" fontWeight={700}>
            <Trans>Delete Collaborator</Trans>
          </ModalHeader>
          <ModalCloseButton />
          <ModalBody>
            <Text>
              <Trans>
                Delete the following collaborator from the VASP account. The collaborator will no
                longer have access to the account.
              </Trans>
            </Text>
            <VStack align="start" spacing={4} mt={2}>
              <Box>
                <Text fontWeight={700}>
                  <Trans>Collaborator Name & Email</Trans>
                </Text>
                <Text textTransform="capitalize">Eason Yang</Text>
                <Text>eyang@vaspnet.co.uk</Text>
              </Box>
              <Checkbox defaultChecked>
                <Trans>Check to delete user account.</Trans>
              </Checkbox>
            </VStack>
          </ModalBody>

          <ModalFooter display="flex" flexDir="column" gap={3}>
            <Button bg="orange" minW="150px" _hover={{ bg: 'orange' }} onClick={onClose}>
              <Trans>Save</Trans>
            </Button>
            <Button variant="ghost" minW="150px" color="link">
              <Trans>Close</Trans>
            </Button>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </Link>
  );
}

export default DeleteCollaboratorModal;
