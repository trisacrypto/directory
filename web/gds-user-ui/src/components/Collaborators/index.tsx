import { Heading, Stack } from "@chakra-ui/react";
import CollaboratorsSection from "components/CollaboratorsSection";

const Collaborators: React.FC = () => {
  return (
    <Stack>
      <Heading marginBottom="69px">Collaborators</Heading>
      <CollaboratorsSection />
    </Stack>
  );
};

export default Collaborators;
