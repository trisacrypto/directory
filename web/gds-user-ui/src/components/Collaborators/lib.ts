import Store from 'application/store';

// map collaborators and set date to date object
export const mapCollaborators = (arr: any) => {
    return arr.map((collaborator: any) => {
        return {
            ...collaborator,
            created_at: new Date(collaborator.created_at),
        };
    });
};

// list by recent date
export const sortCollaboratorsByRecentDate = (arr: any) => {
    const refactoredArr = mapCollaborators(arr);
    return refactoredArr.sort((a: any, b: any) => b.created_at.getTime() - a.created_at.getTime());
};

// is collaborator current user
export const isCurrentUser = (collaboratorEmail: string): boolean => {
    return collaboratorEmail === Store.getState()?.user?.user?.email;
};


