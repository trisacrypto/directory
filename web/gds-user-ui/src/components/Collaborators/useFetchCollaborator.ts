import { useQuery } from '@tanstack/react-query';
import { useToast } from '@chakra-ui/react';

import { getAllCollaborators } from 'modules/dashboard/collaborator/CollaboratorService';
import type { getCollaborators } from 'modules/dashboard/collaborator/getCollaboratorType';

export function useFetchCollaborators(): getCollaborators {
  const toast = useToast();
  const query = useQuery(['fetch-collaborators'], getAllCollaborators, {
    refetchOnWindowFocus: false,
    refetchOnMount: true,
    // set state time to 15 minutes
    staleTime: 1000 * 60 * 15,

    onError: (error: any) => {
      toast({
        title: 'Error',
        description: error?.message,
        status: 'error',
        duration: 5000,
        isClosable: true
      });
    }
  });
  return {
    getAllCollaborators: query.refetch,
    collaborators: query.data?.data?.collaborators,
    hasCollaboratorsFailed: query.isError,
    wasCollaboratorsFetched: query.isSuccess,
    isFetchingCollaborators: query.isLoading,
    error: query.error
  };
}
