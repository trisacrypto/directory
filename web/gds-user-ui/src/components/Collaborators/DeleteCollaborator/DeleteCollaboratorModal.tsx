import {
  Box,
  Button,
  Checkbox,
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
  Text,
  useDisclosure,
  VStack,
  useToast
} from '@chakra-ui/react';
import { useFetchCollaborators } from 'components/Collaborators/useFetchCollaborator';
import { useDeleteCollaborator } from 'components/Collaborators/DeleteCollaborator/useDeleteCollaborator';
import { Trans } from '@lingui/react';
import { BsTrash } from 'react-icons/bs';
import { useEffect, useState } from 'react';
import type { Collaborator } from 'components/Collaborators/CollaboratorType';
import { t } from '@lingui/macro';
import { hasPermission } from 'utils/permission';
import { USER_PERMISSION } from 'types/enums';
import { isCurrentUser } from '../lib';
interface Props {
  collaboratorId: string;
}
function DeleteCollaboratorModal(props: Props) {
  const { collaboratorId } = props;
  const { isOpen, onOpen, onClose } = useDisclosure();
  const toast = useToast();
  const { collaborators, getAllCollaborators } = useFetchCollaborators();
  const {
    isDeleting,
    wasCollaboratorDeleted,
    deleteCollaborator,
    hasCollaboratorFailed,
    errorMessage
  } = useDeleteCollaborator();
  const [collaborator, setCollaborator] = useState<Collaborator>();

  const [isDeleteChecked, setIsDeleteChecked] = useState(false);

  const deleteHandler = () => {
    // delete collaborator
    deleteCollaborator(collaboratorId);

    setIsDeleteChecked(false);
    onClose();
  };

  const shouldDisableButton = () => {
    if (isCurrentUser(collaborator?.email || '')) {
      return true;
    }
    return hasPermission(USER_PERMISSION.UPDATE_COLLABORATOR);
  };

  useEffect(() => {
    if (wasCollaboratorDeleted) {
      getAllCollaborators();
      // display success toast
      toast({
        title: 'Collaborator deleted',
        description: 'The collaborator has been deleted',
        status: 'success',
        duration: 9000,
        isClosable: true,
        position: 'top-right'
      });
    }
  }, [wasCollaboratorDeleted, getAllCollaborators, toast]);

  useEffect(() => {
    const col = collaborators?.find((c: Collaborator) => c.id === collaboratorId);
    if (col) {
      setCollaborator(col);
    }
  }, [collaboratorId, collaborators]);

  useEffect(() => {
    if (hasCollaboratorFailed && !wasCollaboratorDeleted) {
      // display error toast
      const hasErrored =
        errorMessage &&
        t`An error occurred while deleting the collaborator, please try again or contact support at support@trisa.io`;
      toast({
        title: t`Collaborator not deleted`,
        description: hasErrored || t`The collaborator has not been deleted`,
        status: 'error',
        duration: 9000,
        isClosable: true,
        position: 'top-right'
      });
    }
  }, [hasCollaboratorFailed, toast, wasCollaboratorDeleted, errorMessage]);

  return (
    <Button
      color="blue"
      onClick={onOpen}
      bg={'transparent'}
      disabled={shouldDisableButton()}
      _hover={{
        bg: 'transparent'
      }}
      _focus={{
        bg: 'transparent'
      }}>
      <BsTrash fontSize="26px" />
      <Modal isOpen={isOpen} onClose={onClose}>
        <ModalOverlay />
        <ModalContent>
          <ModalHeader textTransform="capitalize" textAlign="center" fontWeight={700}>
            <Trans id="Delete Collaborator">Delete Collaborator</Trans>
          </ModalHeader>
          <ModalCloseButton />
          <ModalBody>
            <Text>
              <Trans id="Delete the following collaborator from the VASP account. The collaborator will no longer have access to the account.">
                Delete the following collaborator from the VASP account. The collaborator will no
                longer have access to the account.
              </Trans>
            </Text>
            <VStack align="start" spacing={4} mt={2}>
              <Box>
                <Text fontWeight={700}>
                  <Trans id="Collaborator Name & Email">Collaborator Name & Email</Trans>
                </Text>
                <Text textTransform="capitalize" data-testid="collaborator-name">
                  {collaborator?.name}
                </Text>
                <Text data-testid="collaborator-email">{collaborator?.email}</Text>
              </Box>
              <Checkbox
                isChecked={isDeleteChecked}
                colorScheme="gray"
                onChange={(e) => setIsDeleteChecked(e.target.checked)}>
                <Trans id="Check to delete user account.">Check to delete user account.</Trans>
              </Checkbox>
            </VStack>
          </ModalBody>

          <ModalFooter display="flex" flexDir="column" gap={3}>
            <Button
              bg="orange"
              minW="150px"
              data-testid="delete-collaborator-button"
              _hover={{ bg: 'orange' }}
              onClick={deleteHandler}
              isDisabled={!isDeleteChecked}
              isLoading={isDeleting}>
              <Trans id="Save">Save</Trans>
            </Button>
            <Button variant="ghost" minW="150px" color="link" onClick={onClose}>
              <Trans id="Close">Close</Trans>
            </Button>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </Button>
  );
}

export default DeleteCollaboratorModal;
