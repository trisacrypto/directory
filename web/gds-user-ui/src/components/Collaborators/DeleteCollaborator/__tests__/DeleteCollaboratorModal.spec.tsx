/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import { fireEvent, screen } from '@testing-library/react';
import { useMutation } from '@tanstack/react-query';
import userEvent from '@testing-library/user-event';
import { dynamicActivate } from 'utils/i18nLoaderHelper';
import nock from 'nock';
import DeleteCollaboratorModal from '../DeleteCollaboratorModal';
import { act, render } from 'utils/test-utils';
import { collaboratorMockValue } from 'components/AddCollaboratorModal/__mocks__';
import * as useCollaborators from 'components/Collaborators/useFetchCollaborator';
import * as useDeleteCollaborator from '../useDeleteCollaborator';
const mockUseMutation = useMutation as jest.Mock;
// mock chakra ui modal component to be able to test it

const divWithChildrenMock = (children: any, identifier: any) => (
  <div data-testId={identifier}>{children}</div>
);
const divWithoutChildrenMock = (identifier: any) => <div data-testId={identifier} />;

jest.mock('@chakra-ui/react', () => ({
  ...jest.requireActual('@chakra-ui/react'),
  Modal: jest.fn(({ children }) => divWithChildrenMock(children, 'modal')),
  ModalOverlay: jest.fn(({ children }) => divWithChildrenMock(children, 'overlay')),
  ModalContent: jest.fn(({ children }) => divWithChildrenMock(children, 'content')),
  ModalHeader: jest.fn(({ children }) =>
    divWithChildrenMock(children, 'delete-collaborator-modal')
  ),
  ModalFooter: jest.fn(({ children }) => divWithChildrenMock(children, 'footer')),
  ModalBody: jest.fn(({ children }) => divWithChildrenMock(children, 'body')),
  ModalCloseButton: jest.fn(() => divWithoutChildrenMock('close'))
}));

// mock deletecollaborator component render function
function renderComponent() {
  const Props = {
    collaboratorId: '1'
  };
  return render(<DeleteCollaboratorModal {...Props} />);
}

const mockCollaborators = jest.fn();
const mockGetAllCollaborators = jest.fn();
const mockDeleteCollaborator = jest.fn();

const useFetchCollaboratorsMock = jest.spyOn(useCollaborators, 'useFetchCollaborators');
const useDeleteCollaboratorMock = jest.spyOn(useDeleteCollaborator, 'useDeleteCollaborator');
describe('DeleteCollaboratorModal', () => {
  beforeAll(() => {
    act(() => {
      dynamicActivate('en');
    });
    useFetchCollaboratorsMock.mockReturnValue({
      collaborators: {
        data: {
          collaborators: collaboratorMockValue.data
        },
        getAllCollaborators: mockGetAllCollaborators
      },
      getAllCollaborators: jest.fn(),
      hasCollaboratorsFailed: false,
      wasCollaboratorsFetched: false,
      isFetchingCollaborators: false
    });
    useDeleteCollaboratorMock.mockReturnValue({
      isDeleting: false,
      wasCollaboratorDeleted: false,
      deleteCollaborator: mockDeleteCollaborator,
      hasCollaboratorFailed: false,
      errorMessage: '',
      reset(): void {
        throw new Error('Function not implemented.');
      }
    });
  });

  it('should render', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render the modal', () => {
    renderComponent();
    expect(screen.getByTestId('delete-collaborator-modal')).toBeInTheDocument();
  });

  it('useCollaborators should be called', () => {
    renderComponent();
    expect(useFetchCollaboratorsMock).toHaveBeenCalled();
  });

  it('useDeleteCollaborator should be called', () => {
    renderComponent();
    expect(useDeleteCollaboratorMock).toHaveBeenCalled();
  });

  it('should show collaborator email in the modal', () => {
    renderComponent();
    expect(screen.getByTestId('collaborator-email').textContent).toBe(
      collaboratorMockValue.data[0].email
    );
  });

  it('should show collaborator name in the modal', () => {
    renderComponent();
    expect(screen.getByTestId('collaborator-name').textContent).toBe(
      collaboratorMockValue.data[0].name
    );
  });

  // delete collaborator click event test
  // it('should call deleteHandler function when delete button is clicked', () => {
  //   renderComponent();
  //   fireEvent.click(screen.getByTestId('delete-collaborator-button'));
  //
  //   expect(mockDeleteCollaborator).toHaveBeenCalled();
  // });

  afterAll(() => {
    jest.clearAllMocks();
  });
});
