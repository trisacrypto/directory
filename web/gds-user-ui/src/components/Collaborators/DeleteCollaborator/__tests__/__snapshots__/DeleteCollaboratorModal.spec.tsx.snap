// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteCollaboratorModal should render 1`] = `
.emotion-0 {
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-fast);
  transition-timing-function: var(--chakra-transition-easing-ease-out);
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 2px solid transparent;
  outline-offset: 2px;
  color: blue;
}

.emotion-0:hover,
.emotion-0[data-hover] {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus,
.emotion-0[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: var(--chakra-space-2);
}

.emotion-2>*:not(style)~*:not(style) {
  margin-top: var(--chakra-space-4);
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-4 {
  font-weight: 700;
}

.emotion-5 {
  text-transform: capitalize;
}

.emotion-7 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: top;
  position: relative;
}

.emotion-7[disabled],
.emotion-7[aria-disabled=true],
.emotion-7[data-disabled] {
  cursor: not-allowed;
}

.emotion-8 {
  width: var(--chakra-sizes-4);
  transition-property: box-shadow;
  transition-duration: var(--chakra-transition-duration-normal);
  border: 2px solid;
  border-radius: var(--chakra-radii-sm);
  border-color: inherit;
  color: var(--chakra-colors-white);
  height: var(--chakra-sizes-4);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  vertical-align: top;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-8[aria-checked=true],
.emotion-8[data-checked] {
  background: var(--chakra-colors-gray-500);
  border-color: var(--chakra-colors-gray-500);
  color: var(--chakra-colors-white);
}

.emotion-8[aria-checked=true]:hover,
.emotion-8[data-checked]:hover,
.emotion-8[aria-checked=true][data-hover],
.emotion-8[data-checked][data-hover] {
  background: var(--chakra-colors-gray-600);
  border-color: var(--chakra-colors-gray-600);
}

.emotion-8[aria-checked=true][disabled],
.emotion-8[data-checked][disabled],
.emotion-8[aria-checked=true][aria-disabled=true],
.emotion-8[data-checked][aria-disabled=true],
.emotion-8[aria-checked=true][data-disabled],
.emotion-8[data-checked][data-disabled] {
  border-color: var(--chakra-colors-gray-200);
  background: var(--chakra-colors-gray-200);
  color: var(--chakra-colors-gray-500);
}

.emotion-8:indeterminate,
.emotion-8[aria-checked=mixed],
.emotion-8[data-indeterminate] {
  background: var(--chakra-colors-gray-500);
  border-color: var(--chakra-colors-gray-500);
  color: var(--chakra-colors-white);
}

.emotion-8[disabled],
.emotion-8[aria-disabled=true],
.emotion-8[data-disabled] {
  background: var(--chakra-colors-gray-100);
  border-color: var(--chakra-colors-gray-100);
}

.emotion-8:focus,
.emotion-8[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-8[aria-invalid=true],
.emotion-8[data-invalid] {
  border-color: var(--chakra-colors-red-500);
}

.emotion-9 {
  -webkit-margin-start: 0.5rem;
  margin-inline-start: 0.5rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: var(--chakra-fontSizes-md);
}

.emotion-9[disabled],
.emotion-9[aria-disabled=true],
.emotion-9[data-disabled] {
  opacity: 0.4;
}

.emotion-10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-10);
  min-width: 150px;
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  background: orange;
}

.emotion-10:focus,
.emotion-10[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-10[disabled],
.emotion-10[aria-disabled=true],
.emotion-10[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-10:hover,
.emotion-10[data-hover] {
  background: orange;
}

.emotion-10:active,
.emotion-10[data-active] {
  background: var(--chakra-colors-gray-300);
}

.emotion-11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-10);
  min-width: 150px;
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  color: link;
}

.emotion-11:focus,
.emotion-11[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-11[disabled],
.emotion-11[aria-disabled=true],
.emotion-11[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-11:hover,
.emotion-11[data-hover] {
  background: var(--chakra-colors-gray-100);
}

.emotion-11:hover[disabled],
.emotion-11[data-hover][disabled],
.emotion-11:hover[aria-disabled=true],
.emotion-11[data-hover][aria-disabled=true],
.emotion-11:hover[data-disabled],
.emotion-11[data-hover][data-disabled] {
  background: initial;
}

.emotion-11:active,
.emotion-11[data-active] {
  background: var(--chakra-colors-gray-200);
}

<div>
  <a
    class="chakra-link emotion-0"
  >
    <svg
      fill="currentColor"
      font-size="26px"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6z"
      />
      <path
        d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1zM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118zM2.5 3V2h11v1h-11z"
        fill-rule="evenodd"
      />
    </svg>
    <div
      data-testid="modal"
    >
      <div
        data-testid="overlay"
      />
      <div
        data-testid="content"
      >
        <div
          data-testid="delete-collaborator-modal"
        >
          Delete Collaborator
        </div>
        <div
          data-testid="close"
        />
        <div
          data-testid="body"
        >
          <p
            class="chakra-text emotion-1"
          >
            Delete the following collaborator from the VASP account. The collaborator will no longer have access to the account.
          </p>
          <div
            class="chakra-stack emotion-2"
          >
            <div
              class="emotion-1"
            >
              <p
                class="chakra-text emotion-4"
              >
                Collaborator Name & Email
              </p>
              <p
                class="chakra-text emotion-5"
                data-testid="collaborator-name"
              >
                test
              </p>
              <p
                class="chakra-text emotion-1"
                data-testid="collaborator-email"
              >
                test@local.dev
              </p>
            </div>
            <label
              class="chakra-checkbox emotion-7"
            >
              <input
                class="chakra-checkbox__input"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control emotion-8"
              />
              <span
                class="chakra-checkbox__label emotion-9"
              >
                Check to delete user account.
              </span>
            </label>
          </div>
        </div>
        <div
          data-testid="footer"
        >
          <button
            class="chakra-button emotion-10"
            data-testid="delete-collaborator-button"
            disabled=""
            type="button"
          >
            Save
          </button>
          <button
            class="chakra-button emotion-11"
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </a>
</div>
`;
